CONTRIBUTING.md

```diff
@@ -0,0 +1,147 @@
+Thanks!
+
+
+
+## Setting Up a Development Environment
+
+**Note**: You may need to substitute ``python3`` for ``python`` in the commands below on some Linux distros where ``python`` isn't mapped to ``python3`` (yet).
+
+
+### Fork and clone the repo
+
+First, navigate to the [project repository](https://github.com/spyder-ide/spyder-docs-sphinx-theme) in your web browser and press the ``Fork`` button to make a personal copy of the repository on your own Github account.
+Then, click the ``Clone or Download`` button on your repository, copy the link and run the following on the command line to clone the repo with submodules:
+
+```bash
+git clone --recursive <LINK-TO-YOUR-REPO>
+```
+
+Finally, set the upstream remote to the official website repo with:
+
+```bash
+git remote add upstream https://github.com/spyder-ide/spyder-docs-sphinx-theme.git
+```
+
+
+### Create and activate a fresh venv
+
+Particularly for development installs, we highly recommend you create and activate a virtual environment to avoid any conflicts with other packages on your system or causing any other issues.
+To do so with Conda (recommended), simply execute the following:
+
+```bash
+conda create -c conda-forge -n theme-env python=3
+```
+
+And activate it with
+
+```bash
+conda activate theme-env
+```
+
+With pip/venv, you can create a virtual environment with
+
+```bash
+python -m venv theme-env
+```
+
+And activate it with the following on Linux and macOS,
+
+```bash
+source theme-env/bin/activate
+```
+
+or on Windows (cmd),
+
+```cmd
+.\theme-env\Scripts\activate.bat
+```
+
+Regardless of the tool you use, make sure to remember to always activate your environment before using it.
+
+
+### Install the development dependencies
+
+To get the consistent pinned versions of the development dependencies, install the deps from the requirements file.
+If using conda, run:
+
+```bash
+conda install -c conda-forge --file docs/requirements.txt
+```
+
+Or, using pip,
+
+```bash
+python -m pip -r docs/requirements.txt
+```
+
+
+### Installing the theme package itself
+
+Finally, install the theme itself in editable mode, along with its ``npm`` dependencies (for rebuilding any CSS/JS files with Webpack; you'll need NodeJS/npm installed and accessable on your path):
+
+```bash
+pip install -e .
+npm install
+```
+
+
+
+## Building and testing the theme
+
+In the root of this repository, to run a continuously updating local webserver with the theme updated interactively, execute ``npm run build:dev`` to build a continually updating test site that will allow you to preview your changes as you make them.
+Once you're satisfied, you can use ``npm run build:production`` to build the theme package for use on the actual docs repo, and finally build the docs as normal (e.g. ``make html``) in the same environment to observe the changes using the updated theme.
```

Comment:
Maybe we need to be more clear about the usage of the theme. Putting as an example spyder-docs here could help understand better the way the theme is used with the production command but I'm not sure.

Reply:
I thought about that, and tried to see how one could work, but after exploring it I'm not sure how much it would really benefit.

The only thing an example would really add is whatever command is used to build the docs in question—which even for our own docs, will vary depending on whether the user wants to build one version or all of them, what platform they are on, etc., which is much better described in our docs contributing guide, whereas the docs theme contains a built-in example (which the current example command works for).

Furthermore, if and when we make changes to the docs build procedure (as is planned), this copy will be out of date here, as one wouldn't expect to have to update a dependency's docs to reflect changes in a dependent, vs. the other way around. So I'd suggest not doing this and keeping the downstream docs build instructions in one canonical source (which per discussions with Juanita and Carlos we plan to modify, revise and simplify).